Intermittent test failures when executing ThrottledTaskRunnerTest

The build fails from time to time due a failure within `ThrottledTaskRunnerTest`

```
INFO] Running com.adobe.acs.commons.fam.impl.ThrottledTaskRunnerTest
10:14:24.627 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very low priority: 0
10:14:24.634 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very high priority: 0
10:14:24.639 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very high priority: 1
10:14:24.640 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very high priority: 2
10:14:24.640 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very high priority: 3
10:14:24.641 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very high priority: 4
10:14:24.641 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very high priority: 5
10:14:24.641 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very high priority: 6
10:14:24.641 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very high priority: 7
10:14:24.642 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very high priority: 8
10:14:24.642 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very high priority: 9
10:14:24.642 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very low priority: 1
10:14:24.642 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very low priority: 2
10:14:24.642 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very low priority: 3
10:14:24.643 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very low priority: 4
10:14:24.643 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very low priority: 5
10:14:24.643 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very low priority: 6
10:14:24.646 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very low priority: 7
10:14:24.646 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very low priority: 8
10:14:24.646 [pool-712-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - very low priority: 9
10:14:25.627 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 0
10:14:25.632 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 1
10:14:25.632 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 2
10:14:25.633 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 3
10:14:25.633 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 4
10:14:25.636 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 5
10:14:25.642 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 6
10:14:25.644 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 7
10:14:25.645 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 8
10:14:25.645 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 9
10:14:25.646 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - normal priority: 0
10:14:25.648 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 10
10:14:25.648 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 11
10:14:25.655 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 12
10:14:25.656 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 13
10:14:25.656 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 14
10:14:25.656 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 15
10:14:25.657 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 16
10:14:25.657 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 17
10:14:25.657 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 18
10:14:25.657 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 19
10:14:25.657 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 20
10:14:25.658 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 21
10:14:25.658 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 22
10:14:25.658 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 23
10:14:25.658 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 24
10:14:25.658 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 25
10:14:25.658 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 26
10:14:25.658 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 27
10:14:25.658 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 28
10:14:25.658 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 29
10:14:25.659 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 30
10:14:25.659 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 31
10:14:25.659 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 32
10:14:25.659 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 33
10:14:25.659 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 34
10:14:25.659 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 35
10:14:25.659 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 36
10:14:25.659 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 37
10:14:25.661 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 38
10:14:25.661 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 39
10:14:25.671 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - high priority: 0
10:14:25.672 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - high priority: 1
10:14:25.673 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - high priority: 2
10:14:25.674 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - high priority: 3
10:14:25.674 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - high priority: 4
10:14:25.675 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - high priority: 5
10:14:25.675 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - high priority: 6
10:14:25.675 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - high priority: 7
10:14:25.676 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - high priority: 8
10:14:25.679 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - high priority: 9
10:14:25.680 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - normal priority: 1
10:14:25.680 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - normal priority: 2
10:14:25.681 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - normal priority: 3
10:14:25.681 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - normal priority: 4
10:14:25.681 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - normal priority: 5
10:14:25.682 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - normal priority: 6
10:14:25.682 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - normal priority: 7
10:14:25.682 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 40
10:14:25.682 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 41
10:14:25.682 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 42
10:14:25.683 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 43
10:14:25.683 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 44
10:14:25.683 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 45
10:14:25.683 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - normal priority: 8
10:14:25.683 [pool-720-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - normal priority: 9
10:14:25.685 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 46
10:14:25.685 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 47
10:14:25.685 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 48
10:14:25.685 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 49
10:14:25.686 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 50
10:14:25.686 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 51
10:14:25.686 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 52
10:14:25.686 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 53
10:14:25.686 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 54
10:14:25.687 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 55
10:14:25.687 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 56
10:14:25.687 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 57
10:14:25.687 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 58
10:14:25.687 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 59
10:14:25.687 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 60
10:14:25.687 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 61
10:14:25.688 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 62
10:14:25.688 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 63
10:14:25.688 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 64
10:14:25.688 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 65
10:14:25.688 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 66
10:14:25.688 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 67
10:14:25.689 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 68
10:14:25.689 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 69
10:14:25.689 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 70
10:14:25.689 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 71
10:14:25.689 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 72
10:14:25.690 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 73
10:14:25.690 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 74
10:14:25.690 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 75
10:14:25.691 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 76
10:14:25.691 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 77
10:14:25.691 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 78
10:14:25.691 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 79
10:14:25.693 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 80
10:14:25.693 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 81
10:14:25.693 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 82
10:14:25.694 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 83
10:14:25.694 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 84
10:14:25.694 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 85
10:14:25.694 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 86
10:14:25.694 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 87
10:14:25.695 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 88
10:14:25.695 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 89
10:14:25.695 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 90
10:14:25.695 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 91
10:14:25.695 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 92
10:14:25.695 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 93
10:14:25.696 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 94
10:14:25.696 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 95
10:14:25.696 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 96
10:14:25.697 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 97
10:14:25.703 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 98
10:14:25.703 [pool-716-thread-1] INFO  c.a.a.c.f.i.ThrottledTaskRunnerTest - Executed job: 99
[ERROR] Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.035 s <<< FAILURE! - in com.adobe.acs.commons.fam.impl.ThrottledTaskRunnerTest
[ERROR] assertFifoOrder(com.adobe.acs.commons.fam.impl.ThrottledTaskRunnerTest)  Time elapsed: 0.016 s  <<< FAILURE!
java.lang.AssertionError: wrong number of items executed expected:<100> but was:<1>
	at com.adobe.acs.commons.fam.impl.ThrottledTaskRunnerTest.assertFifoOrder(ThrottledTaskRunnerTest.java:152)
...
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ThrottledTaskRunnerTest.assertFifoOrder:152 wrong number of items executed expected:<100> but was:<1>
[INFO] 
[ERROR] Tests run: 1432, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] ACS AEM Commons - Reactor Project .................. SUCCESS [  0.800 s]
[INFO] ACS AEM Commons Bundle ............................. FAILURE [02:59 min]
[INFO] ACS AEM Commons Oakpal Checks ...................... SKIPPED
[INFO] ACS AEM Commons Package ............................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

```

This can be seen e.g. in https://api.travis-ci.org/v3/job/543661231/log.txt. The current master branch is affected, but it only fails once in a while.
Maybe some kind of race condition.

--- a/bundle/src/main/java/com/adobe/acs/commons/fam/impl/ThrottledTaskRunnerImpl.java
+++ b/bundle/src/main/java/com/adobe/acs/commons/fam/impl/ThrottledTaskRunnerImpl.java
@@ -73,7 +73,7 @@ public class ThrottledTaskRunnerImpl extends AnnotatedStandardMBean implements T
     private int maxThreads;
     private double maxCpu;
     private double maxHeap;
-    private boolean isPaused;
+    private volatile boolean isPaused;
     private final MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
     private ObjectName osBeanName;
     private ObjectName memBeanName;

--- a/bundle/src/test/java/com/adobe/acs/commons/fam/impl/ThrottledTaskRunnerTest.java
+++ b/bundle/src/test/java/com/adobe/acs/commons/fam/impl/ThrottledTaskRunnerTest.java
@@ -66,6 +66,8 @@ public class ThrottledTaskRunnerTest {
 
         ttr.resumeExecution();
 
+        // Allow the threadpool to start in the background
+        Thread.sleep(100);
         while (ttr.getActiveCount() > 0) {
             Thread.sleep(1000);
         }
@@ -109,6 +111,8 @@ public class ThrottledTaskRunnerTest {
 
         ttr.resumeExecution();
 
+        // Allow the threadpool to start in the background
+        Thread.sleep(100);
         while (ttr.getActiveCount() > 0) {
             Thread.sleep(1000);
         }
@@ -143,8 +147,11 @@ public class ThrottledTaskRunnerTest {
             }, Integer.MIN_VALUE);
         }
 
+
         ttr.resumeExecution();
 
+        // Allow the threadpool to start in the background
+        Thread.sleep(100);
         while (ttr.getActiveCount() > 0) {
             Thread.sleep(1000);
         }

